test/Npgsql.Tests/CopyTests.cs

```diff
@@ -738,6 +738,19 @@ public async Task Cancel_binary_export_when_not_consumed_and_then_Dispose()
         Assert.That(async () => await conn.ExecuteScalarAsync("SELECT 1"), Is.EqualTo(1), "The connection is still OK");
     }
 
+    [Test]
+    [IssueLink("https://github.com/npgsql/npgsql/issues/4417")]
+    public async Task Binary_copy_throws_for_nullable()
+    {
+        await using var conn = await OpenConnectionAsync();
+        await using var _ = await CreateTempTable(conn, "house_number integer", out var tableName);
+
+        await using var writer = await conn.BeginBinaryImportAsync($"COPY {tableName}(house_number) FROM STDIN BINARY");
+        int? value = 1;
+        await writer.StartRowAsync();
+        var ex = Assert.ThrowsAsync<InvalidCastException>(async () => await writer.WriteAsync(value, NpgsqlDbType.Integer))!;
```

Comment:
Unless you want to assert something on the exception...

```suggestion
        Assert.ThrowsAsync<InvalidCastException>(async () => await writer.WriteAsync(value, NpgsqlDbType.Integer));
```

Reply:
Ah, missed that - was thinking about checking whether the message is correct...